src/main/java/com/conveyal/r5/streets/EdgeStore.java

```diff
@@ -728,6 +729,32 @@ public boolean canTurnFrom(StreetRouter.State s0, StreetRouter.State s1,
             return true;
         }
 
+        /**
+         * Cut the packed list of intermediate coordinates in two after specified segment
```

Comment:
I think the Javadoc should be revised slightly. Because the arrays store only the intermediate (non-end) points of the geometry, this operation can be tricky to think about and a little more detail could save some confusion in the future. Especially the expression "coordinates after end of specified segment" gave me pause, since I believe the second return value includes the end of the specified segment. I thought through the logic and this method seems correct to me. Here's my attempt at revised Javadoc: 

Cut the packed array of intermediate coordinates in two at the specified segment index. The intermediate coordinates array does not include the first and last point of the geometry (which are supplied by vertices) so the segment at index N ends at intermediate point N, which occupies array indexes N*2 and N*2+1 in the packed array. All the points before the specified segment (including the segment's beginning point) should be the first return array, and all the points after the specified segment (including the segment's ending point) should be in the second return array. The first or second return array can be of length zero if we are splitting at the first or last segment respectively.

Reply:
It's also worth noting in the Javadoc that this is a nondestructive split, the original geometry is left as-is and copies are returned by the method.